examples/register_and_change_username.php

```diff
@@ -43,35 +44,24 @@
     ],
 ];
 
-$dispatch = Kernel\buildCommandDispatcher(
-    $commandMap,
-    $factories['eventStore'],
-    $factories['snapshotStore']
-);
+$dispatch = Kernel\buildCommandDispatcher($factories['eventStore'](), $commandMap, $factories['snapshotStore']());
 
-$command = new RegisterUser(['id' => '1', 'name' => 'Alex', 'email' => 'member@getprooph.org']);
+/* @var Validation $result */
+$result = $dispatch(new RegisterUser(['id' => '1', 'name' => 'Alex', 'email' => 'member@getprooph.org']));
+echo $result->show() . PHP_EOL;
+echo json_encode($result->getOrElse('')->head()->payload()) . PHP_EOL . PHP_EOL;
```

Comment:
Is this going to blow when the result is a `Failure`?
Perhaps what you want is:
```php
use Phunkie\PatternMatching\Referenced\Success;

$on = match($result); switch(true) {
    case $on(Success($a)): echo $a->head()->payload() .
        PHP_EOL . PHP_EOL; break;
    case $on(Failure(_)): echo PHP_EOL . PHP_EOL; break;
}
```